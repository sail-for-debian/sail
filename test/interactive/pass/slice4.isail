:slice_roots k
:thin_slice
:ast out.sail
:quit
