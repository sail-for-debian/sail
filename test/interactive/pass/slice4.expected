default Order dec







































$sail_internal 






enum e2 = {GHI, JKL}

struct s1 = {fA : int, fB : e2}

struct s3 = {fE : s1, fF : int}

val undefined_s3 : unit -> s3

val f : int -> int

val k : int -> int

function k x = {
    var s : s3 = undefined_s3();
    s = { s with fF = x };
    f(s.fF)
}
